server/app/controllers/applicant/ApplicantProgramBlocksController.java

```diff
@@ -513,32 +529,22 @@ private CompletionStage<Result> renderErrorOrRedirectToNextBlock(
               .createAddressQuestion();
 
       if (addressQuestion.needsAddressCorrection()) {
+
         return applicantService
             .getAddressSuggestionGroup(thisBlockUpdated)
-            .thenApplyAsync(
+            .thenComposeAsync(
                 addressSuggestionGroup -> {
-                  ImmutableList<AddressSuggestion> suggestions =
-                      addressSuggestionGroup.getAddressSuggestions();
-                  String json = addressSuggestionJsonSerializer.serialize(suggestions);
-
-                  Boolean isEligibilityEnabledOnThisBlock =
-                      thisBlockUpdated.getLeafAddressNodeServiceAreaIds().isPresent();
-
-                  return ok(addressCorrectionBlockView.render(
-                          buildApplicationBaseViewParams(
-                              request,
-                              applicantId,
-                              programId,
-                              blockId,
-                              inReview,
-                              roApplicantProgramService,
-                              thisBlockUpdated,
-                              applicantName,
-                              ApplicantQuestionRendererParams.ErrorDisplayMode.DISPLAY_ERRORS),
-                          messagesApi.preferred(request),
-                          addressSuggestionGroup,
-                          isEligibilityEnabledOnThisBlock))
-                      .addingToSession(request, ADDRESS_JSON_SESSION_KEY, json);
+                  return maybeRenderAddressCorrectionScreen(
```

Comment:
I think you might not need the `{}` and `return` here if it's only one statement?